src/AutoEnrollPlugin/Service/AutoEnrollmentService.cs

```diff
@@ -0,0 +1,195 @@
+        }
+
+        public void Start()
+        {
+            this.actionBlock = new ActionBlock<Notification>(async notification =>
+            {
+                try
+                {
+                    var mappings = streamMappingService.CreateMappings(notification.StreamId);
+
+                    this.logger.Debug("Found {mappings} mappings for stream {stream}", mappings?.Count, notification.StreamId);
+
+                    foreach (var mapping in mappings)
+                    {
+                        var isValidationPassed = validationService.Validate(notification, mapping);
+
+                        if (isValidationPassed)
+                        {
+                            var isBlocked = this.debouncingService.IsBlocked(notification, mapping);
+
+                            if (isBlocked)
+                            {
+                                continue;
+                            }
+
+                            this.debouncingService.Block(notification, mapping);
+
+                            await enrollAsync(notification, mapping);
+                        }
+                    }
+                }
+                catch (Exception ex)
+                {
+                    this.logger.Error(ex, "Failed to process message");
+                }
+            },
+            new ExecutionDataflowBlockOptions
+            {
+                MaxDegreeOfParallelism = MAX_PARALLEL_BLOCKS
+            });
+        }
+
+        public async Task StopAsync()
+        {
+            actionBlock.Complete();
+            await actionBlock.Completion;
+        }
+
+        public void ProcessNotification(Notification notification)
+        {
+            if (notification == null)
+            {
+                throw new ArgumentNullException(nameof(notification));
+            }
+
+            this.actionBlock.Post(notification);
+        }
+
+        private async Task enrollAsync(Notification notification, StreamMapping mapping)
+        {
+            if (notification == null)
+            {
+                throw new ArgumentNullException(nameof(notification));
+            }
+
+            if (mapping == null)
+            {
+                throw new ArgumentNullException(nameof(mapping));
+            }
+
+            this.logger.Information("Enrolling new member to watchlist {watchlist}", mapping.WatchlistIds);
+
+            if (!(mapping.WatchlistIds?.Length > 0))
+            {
+                this.logger.Information("No target watchlist id, skipped");
+                return;
+            }
+
+            var schema = this.configuration.GetValue<string>("Target:Schema", "http");
+            var host = this.configuration.GetValue<string>("Target:Host", "SFApi");
+            var port = this.configuration.GetValue<int>("Target:Port", 8098);
+
+            var baseUri = new Uri($"{schema}://{host}:{port}/");
+
+            var httpClient = this.httpClientFactory.CreateClient();
```

Comment:
creating HttpClient per each notification is not a best practice, but dispose this client if so -> `using var httpClient =`